test/lit/merge/renamings.wat

```diff
@@ -0,0 +1,360 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: wasm-merge %s first %s.second second -all -S -o - | filecheck %s
+
+;; Test that we rename items in the second module to avoid name collisions.
+
+(module
+  ;; CHECK:      (type $array (array (mut funcref)))
+  (type $array (array (mut (ref null func))))
+
+  ;; This tag has a conflict in second.wat, and so second.wat's $foo
+  ;; will be renamed.
+  ;; CHECK:      (type $none_=>_none (func))
+
+  ;; CHECK:      (type $ref|$array|_=>_none (func (param (ref $array))))
+
+  ;; CHECK:      (type $i32_=>_none (func (param i32)))
+
+  ;; CHECK:      (type $i64_=>_none (func (param i64)))
+
+  ;; CHECK:      (type $f32_=>_none (func (param f32)))
+
+  ;; CHECK:      (type $f64_=>_none (func (param f64)))
+
+  ;; CHECK:      (global $foo i32 (i32.const 1))
+
+  ;; CHECK:      (global $bar i32 (i32.const 2))
+
+  ;; CHECK:      (global $other i32 (i32.const 3))
+
+  ;; CHECK:      (global $bar_2 i32 (i32.const 4))
+
+  ;; CHECK:      (memory $foo 10 20)
+
+  ;; CHECK:      (memory $bar 30 40)
+
+  ;; CHECK:      (memory $foo_2 50 60)
+
+  ;; CHECK:      (memory $other 70 80)
+
+  ;; CHECK:      (data $foo (i32.const 1) "abc")
+
+  ;; CHECK:      (data $bar (i32.const 2) "def")
+
+  ;; CHECK:      (data $other (memory $foo_2) (i32.const 3) "ghi")
+
+  ;; CHECK:      (data $bar_2 (memory $foo_2) (i32.const 4) "jkl")
+
+  ;; CHECK:      (table $foo 10 20 funcref)
+
+  ;; CHECK:      (table $bar 30 40 funcref)
+
+  ;; CHECK:      (table $foo_2 50 60 funcref)
+
+  ;; CHECK:      (table $other 70 80 funcref)
+
+  ;; CHECK:      (elem $foo func $foo $bar)
+
+  ;; CHECK:      (elem $bar func $bar $foo)
+
+  ;; CHECK:      (elem $other func $foo_3 $other)
+
+  ;; CHECK:      (elem $bar_2 func $other $foo_3)
+
+  ;; CHECK:      (tag $foo (param i32))
+  (tag $foo (param i32))
+
+  ;; CHECK:      (tag $bar (param i64))
+  (tag $bar (param i64))
```

Comment:
Where did the tags from the other module go? It looks like they are correctly referenced, but I don't actually see their declarations. Is this a validator bug, too, or am I blind?

Reply:
I'm not sure why they end up sorted to an odd position by the auto-updater, but they are down on lines 102,104.